File: IOI/IOI 10-languages/grader.cpp
#include "grader.h"
#include "lang.h"

#include <bits/stdc++.h>
using namespace std;
#define N 100

static char lang[20], lan[100][20];
static int lnum, i,j,k,n,nl, uni[N], correct, tot;
int wrong[56];
int total[56];
int language(int L) {
   if (L < 0 || L >= 56) exit(92);
   correct += (L == lnum);
   tot++;
   if (L != lnum) {
       wrong[lnum]++;
   }
   total[lnum]++;
   return lnum;
}

int main(){
   for (n=0; 1 == scanf("%s",lang); n++) {
      for (i=0;i<nl && strcmp(lang,lan[i]);i++);
      strcpy(lan[i],lang);
      if (i == nl) {
         nl++;
      }
      //cout << lang << " " << i << endl;
      lnum = i;
      for (i=0;i<N;i++) scanf("%d,",&uni[i]);
      if (lnum == 45) {
        //for (int j = 0; j < 100; j++) cout << uni[j] << " ";
        //cout << endl;
      }
      scanf("%*[^\n]");
      excerpt(uni);
   }
   printf("OK %0.2lf%%\n",100.0*correct/tot);
   /*
   for (int i = 0; i <= 55; i++) {
       cout << "Language " << i << ": " << 100.0*wrong[i]/total[i] << endl;
   }
   */
   return 0;
}

File: IOI/IOI 10-languages/grader.pas
program grader;
   uses gradershare, lang;
   var i,n: longint;
   var x,y: char;
   var lan1: array [0..55] of char;
   var lan2: array [0..55] of char;
begin
   n := 0;
   nl := 0;
   right := 0;
   tot := 0;
   while not eof do begin
      read(x,y);
      for i:=0 to 99 do read(uni[i]);
      readln;
      i := 0;
      while (i < nl) and ((lan1[i] <> x) or (lan2[i]<> y)) do i := i + 1;
      if i = nl then begin
         lan1[i] := x; lan2[i] := y; nl := nl + 1;
      end;
      lnum := i;
      excerpt(uni);
      tot := tot + 1;
   end;
   writeln('OK ',100*right/tot:0:2,'%');
end.
